packages/ogmios/src/ogmiosToCore/tx.ts

```diff
@@ -65,319 +53,305 @@ const mapPoolParameters = (poolParameters: Schema.PoolParameters): Cardano.PoolP
-  key: Crypto.Ed25519PublicKeyHex(b.key!),
-  signature: Crypto.Ed25519SignatureHex(HexBlob.fromBase64(b.signature!))
+  key: Crypto.Ed25519PublicKeyHex(b.key),
+  signature: Crypto.Ed25519SignatureHex(HexBlob.fromBase64(b.signature))
 });
 
 const mapRedeemer = (key: string, redeemer: Schema.Redeemer): Cardano.Redeemer => {
   const purposeAndIndex = key.split(':');
+  const { memory, cpu: steps } = redeemer.executionUnits;
 
   return {
     data: Serialization.PlutusData.fromCbor(HexBlob(redeemer.redeemer)).toCore(),
-    executionUnits: redeemer.executionUnits,
+    executionUnits: { memory, steps },
     index: Number(purposeAndIndex[1]),
     purpose: purposeAndIndex[0] as Cardano.RedeemerPurpose
   };
 };
 
 const mapMetadatum = (obj: Schema.Metadatum): Cardano.Metadatum => {
-  if ('string' in obj) {
-    return obj.string;
-  } else if ('map' in obj) {
-    return new Map(obj.map.map(({ k, v }) => [mapMetadatum(k), mapMetadatum(v)]));
-  } else if ('list' in obj) {
-    return obj.list.map(mapMetadatum);
-  } else if ('int' in obj) {
-    return obj.int;
-  } else if ('bytes' in obj) {
-    return Buffer.from(obj.bytes, 'hex');
+  if (typeof obj === 'string') {
+    return obj;
+  } else if (typeof obj === 'bigint') {
+    return obj;
+  } else if (Array.isArray(obj)) {
+    return obj.map(mapMetadatum);
+  } else if (typeof obj === 'object' && obj !== null) {
+    return new Map(Object.entries(obj).map(([k, v]) => [mapMetadatum(k), mapMetadatum(v)]));
   }
-  throw new NotImplementedError(
-    `Unknown metadatum type: ${typeof obj === 'object' ? Object.keys(obj).join(',') : obj}`
-  );
+  // TODO: deprecate Uint8Array union type from Cardano.Metadatum
+  // 'bytes' type does not exist on Schema.Metadata anymore
+  // else if ('bytes' in obj) {
+  //   return Buffer.from(obj.bytes, 'hex');
+  // }
+  throw new NotImplementedError(`Unknown metadatum type: ${obj}`);
 };
 
 const mapAuxiliaryData = (
-  data: Schema.AuxiliaryData | null
+  data: Transaction['metadata']
 ): { auxiliaryData: Cardano.AuxiliaryData | undefined; auxiliaryDataHash: Crypto.Hash32ByteBase16 | undefined } => {
-  if (data === null) return { auxiliaryData: undefined, auxiliaryDataHash: undefined };
+  if (!data) return { auxiliaryData: undefined, auxiliaryDataHash: undefined };
 
   const auxiliaryData = {
-    blob: data.body.blob
-      ? new Map(Object.entries(data.body.blob).map(([key, value]) => [BigInt(key), mapMetadatum(value)]))
-      : undefined,
-    scripts: data.body.scripts ? data.body.scripts.map(mapScript) : undefined
+    blob: data.labels
+      ? new Map(Object.entries(data.labels).map(([key, value]) => [BigInt(key), mapMetadatum(value)]))
+      : undefined
+    // Extra scripts have been moved to the scripts field and merged with witness scripts.
+    // https://github.com/CardanoSolutions/ogmios/blob/master/architectural-decisions/accepted/017-api-version-6-major-rewrite.md#transaction
+    // TODO: deprecate 'scripts' from auxiliaryData
```

Comment:
TODO: deprecate 'scripts' from auxiliaryData

Reply:
Agree with @AngelCastilloB here. Since there's information loss here we can't determine where it came from. Mapping to witness scripts is fine for our use case for now, but we should probably update our `OnChainTx` model later, because this mapping will be inaccurate representation.

Reply:
Actually it says `The hash digests of those metadata-scripts are now listed as requiredExtraScripts`. Maybe it means that there's no information loss after all and we can check if script hash is present in that field to determine the correct mapping?

Reply:
We can remove the `witness.scripts` and `auxiliaryData.scripts` in `OnChainTx` (keep in `Tx` type) type because we are not using it for anything. We can come back to this if/when we need to do anything with OnChainTx scripts

Reply:
Removed here: [3d41f58](https://github.com/input-output-hk/cardano-js-sdk/pull/927/commits/3d41f5853e0968fd4d33ebca4d4c2d5a5627d429)
@AngelCastilloB, please have a second look at this 😄 .

Reply:
@mkazlauskas can you resolve this on behalf of @AngelCastilloB ?

Reply:
Looks good to me. We may have to come back to this if we ever need to do something with OnChainTx scripts